### ISSUE TITLE
Clarify alertmanager configuration and templates documentation

### ISSUE THREAD

**Issue by slok** [coreos/prometheus-operator#389]
Hi!

I was configuring the altermanager on the operator and I've reach the point that the documentation wasn't very clear.

I started with the alertmanager configuration, I read the [spec](https://github.com/coreos/prometheus-operator/blob/master/Documentation/api.md#alertmanagerspec) and didn't saw where is the alertmanager configuration set, then I went to the [user guides](https://github.com/coreos/prometheus-operator/blob/master/Documentation/user-guides/alerting.md) and I could imagine how to do it, but it's not clear at all, so I started [reading the code](https://github.com/coreos/prometheus-operator/tree/e31a9dc1f309ae7ef1aa479802303e33d69a333e/pkg/alertmanager/statefulset.go#L273-L279) and I saw that the secret needs to be called `alertmanager-{alertmanager_name}`. 

Then it came the templating stuff, my configuration needs templates, so I searched for a place to set a ConfigMap or Secret and I didn't saw anything on the [AlertmanagerSpec](https://github.com/coreos/prometheus-operator/blob/master/Documentation/api.md#alertmanagerspec) neither, I ended up setting it on the same secret resource as the configuration, like this:
```
apiVersion: v1
kind: Secret
metadata:
  name: alertmanager-main
data:
  alertmanager.yaml: ............
  hipchat.tmpl: ............
```
and then using this path for the templates on the alertmanager.conf:
```
templates:
- '*.tmpl'
```

**Comment by brancz** [comment]
I'm not sure I understand the question. Do you mean that the last step doesn't work?

```
templates:
- '*.tmpl'
```

You might need to specify the full path `/etc/alertmanager/config/*.tmpl`.

**Comment by slok** [comment]
Sorry, I didn't explained good enought. 

There is no question, what I wrote works like a charm, I just explained how I managed to get it work with the lack of documentation (or at least not clear enought for me). The issue is only to put some little explanation on the user guide or similar of how the configuration and the templates can be set up correctly with the `Alertmanager` resource.

I could help clarifying this on the user guide of the `Alertmanager` as a PR if you prefer

Thank you!
